### PR TITLE
NO-TICKET - dynamic button offset

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Description of Work

support this thing
Снимок экрана 2023-02-13 в 18 17 01

### CODE REVIEW

**Review by gslevinkov — APPROVED** [review]

File: core/src/main/java/org/p2p/core/utils/DynamicOffset.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            .onCompletion { translationY = 0f }
+            .launchIn(viewScope)
+            .also { job -> view.setTag(tagId, job) }
+    }
+}
+
+fun View.disableOnKeyboardOpenDynamicBottomOffset() {
+    val job = getTag(R.id.view_keyboard_listener_job_tag_id) as? Job ?: return
+    job.cancel()
+}
```
Comment (1sakov) [inline core/src/main/java/org/p2p/core/utils/DynamicOffset.kt:30]: Can we move these in Keyboard or view extensions?

File: core/src/main/java/org/p2p/core/utils/ViewScope.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+private class ViewCoroutineScope : CoroutineScope, View.OnAttachStateChangeListener {
+    override val coroutineContext = SupervisorJob() + Dispatchers.Main
+
+    override fun onViewAttachedToWindow(view: View) = Unit
+
+    override fun onViewDetachedFromWindow(view: View) {
+        coroutineContext.cancel()
+        view.setTag(R.id.view_coroutine_scope_tag_id, null)
+    }
+}
```
Comment (1sakov) [inline core/src/main/java/org/p2p/core/utils/ViewScope.kt:33]: Where do you use it?

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/ViewExtensions.kt [hunk 3]
```diff
@@ -11,18 +21,12 @@ import android.view.ViewGroup
...
-import androidx.core.view.doOnAttach
-import androidx.core.view.doOnDetach
-import androidx.core.view.doOnLayout
-import androidx.fragment.app.Fragment
-import androidx.recyclerview.widget.RecyclerView
+import org.p2p.core.utils.disableOnKeyboardOpenDynamicBottomOffset
+import org.p2p.core.utils.enableOnKeyboardOpenDynamicBottomOffset
 import org.p2p.core.utils.showKeyboard
 
+val ON_KEYBOARD_OPEN_DELTA_OFFSET = 28f.toPx()
```
Comment (1sakov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/ViewExtensions.kt:28]: private?

File: core/src/main/java/org/p2p/core/utils/ViewScope.kt (hunk 2 above)
Comment (NikolaiEmelianov) [reply core/src/main/java/org/p2p/core/utils/ViewScope.kt:33]: https://github.com/p2p-org/key-app-android/pull/1242/files/0cee19e3d66c9e113bde2eecf64359e75928ff20#diff-859b1adba07eca3f759298422afe59a6ea9e82618a9f731a2374d66b7b143175R22

**Review by 1sakov — APPROVED** [review]

**Review by 1sakov — CHANGES_REQUESTED** [review]
do we really need such complex thing for only offset?
Can you ask designers what is the value of it? 
I would just skip this

File: core/src/main/java/org/p2p/core/utils/DynamicOffset.kt (hunk 1 above)
Comment (Davran331122) [reply core/src/main/java/org/p2p/core/utils/DynamicOffset.kt:30]: `translationY = onKeyboardOpenDeltaOffset.takeIf { isVisible } ?: 0f `

File: core/src/main/java/org/p2p/core/utils/ViewScope.kt (hunk 2 above)
Comment (Davran331122) [reply core/src/main/java/org/p2p/core/utils/ViewScope.kt:33]:
```
return storedScope ?: ViewCoroutineScope().apply {
            if(isAttachedToWindow) {
                addOnAttachStateChangeListener(this)
                setTag(R.id.view_coroutine_scope_tag_id, this)
            } else {
                cancel()
            }
        }
```

**Pull request closed without merging**